internal/clients/telegram/state/daily_status.go

```diff
@@ -0,0 +1,235 @@
+
+	return s.Ignore()
+}
+
+/*
+Checks if the query is to save the default project. The retuned bool indicates if the query was about saving the default
+project. If returns false try other queries if present or return an ignore response.
+*/
+func (s *DailyStatusHandler) handleCQSaveDefaultProject(data string, message update.Message) (Transition, bool) {
+	askDiscoveryOfTheDay := []response.BotAction{
+		response.RemoveReplyMarkup(message),
+		response.NewSendMessage(response.ChatID(fmt.Sprint(message.Chat.ID)), s.responses.DiscoveryOfTheDay),
+	}
+
+	if data == cqDailyStatusAskDefaultProjectEveryTime {
+		s.UseOnlyProjectNoSaveDefault = true
+
+		return NewTransition(s.DailyStatusState, s.userData, askDiscoveryOfTheDay), true
+	}
+
+	if data == cqDailyStatusSetOnlyProjectAsDefault {
+		s.DefaultProject = option.Some[github.ProjectID](s.UseProject)
+
+		return NewTransition(s.DailyStatusState, s.userData, askDiscoveryOfTheDay), true
+	}
+
+	return Transition{}, false
+}
+
+func (s *DailyStatusHandler) generateReport() (string, error) {
+	items, err := github.NewClient(s.userData.GithubAPIKey.UnwrapOr("")).ListViewerProjectV2Items(context.Background(),
+		s.UseProject, option.Some(dailyStatusItemLimit), option.None[github.ProjectCursor]())
+	if err != nil {
+		return "", errors.WithMessage(err, "while getting user's project v2 items")
+	}
+
+	const listSep = "\n• "
+
+	report := fmt.Sprintf(`#daily report %s:
+<b>Today I worked on</b>%s
+
+<b>Tomorrow I will work on</b>%s
+
+`, time.Now().Format("01.02"), listSep+strings.Join(items["Done"], listSep),
+		listSep+strings.Join(items["In Progress"], listSep))
+
+	if dod, isSome := s.DiscoveryOfTheDay.Unwrap(); isSome {
+		report += "<b>Discovery of the day</b>\n\n" + dod + "\n\n"
+	}
+
+	if blockers, isSome := s.QuestionsAndBlockers.Unwrap(); isSome {
+		report += "<b>Questions/Blockers</b>\n\n" + blockers + "\n\n"
+	}
+
+	if len(items["In Review"]) != 0 {
+		report += "<b>In review</b>" + listSep + strings.Join(items["In Review"], listSep)
+	}
+
+	return report, nil
+}
+
+type DailyStatusState struct {
+	Stage                dailyStatusStage
+	DiscoveryOfTheDay    option.Option[string]
+	QuestionsAndBlockers option.Option[string]
+	UseProject           github.ProjectID // The project we are generating report for
+	RootState
+}
+
+/*
+Ignores user's messages until they click a button to optionaly save the only project they have as the default for this
+chat.
+*/
+func NewDailyStatusStateAskSaveDefault(state RootState, project github.ProjectID) DailyStatusState {
+	return DailyStatusState{
+		Stage:                ignoreMessagesDailyStatusStage,
+		DiscoveryOfTheDay:    option.None[string](),
+		QuestionsAndBlockers: option.None[string](),
+		UseProject:           project,
+		RootState:            state,
+	}
+}
+
+// The next message in the chat will be the answer for the first question in the sequence.
+func NewDailyStatusStateForProject(state RootState, project github.ProjectID) DailyStatusState {
```

Comment:
DailyStatusState should just use the RootState's default project settings and assume its Some and if none then https://github.com/m-kuzmin/daily-reporter/pull/7/files#r1259775769